maputreexo.go

```diff
@@ -0,0 +1,408 @@
+package utreexo
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/slices"
+)
+
+// MapUtreexo is a Utreexo implementation using a hashmap.
+type MapUtreexo struct {
+	CachedLeaves []uint64
+	Nodes        map[uint64]Hash
```

Comment:
Just an idea: What if instead of map, we use an interface allowing anything that exposes a key-value interface, like `leveldb`. So, we would get a disk Pollard “for free”.  I was thinking in a interface like:
```go
type PollardDatabase interface {
    Save(key uint64, value Node)
    Fetch(key uint64)
    Delete(key uint64)
    Update(key uint64, new_value Node)
}
```

Reply:
Yeah good idea!